include/zephyr/sys/predef_heap.h

```diff
@@ -0,0 +1,166 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * Author: Adrian Warecki <adrian.warecki@intel.com>
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_PREDEF_HEAP_H_
+#define ZEPHYR_INCLUDE_SYS_PREDEF_HEAP_H_
+
+#include <zephyr/types.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/sys/util.h>
+
+/* Simple heap implementation with predefined buffers size.
+ *
+ * The blob of memory passed to the heap will be divided into a predefined number of buffers
+ * of specified sizes. Buffers of various sizes are supported. Configuration of a memory division
+ * for a buffers can be changed in runtime. The allocator tries to allocate the smallest buffer that
+ * satisfies the request. Adjacent buffers are not merged, which eliminates the problem of memory
+ * fragmentation. Allocate and free by user pointer and not an opaque block handle. Current
+ * configuration and bookkeeping is stored in a separate memory.
+ */
```

Comment:
Please use Doxygen ["file" annotation](https://www.doxygen.nl/manual/commands.html#cmdfile) for such descriptions.